Introduce a page on Tenderly (#2069)
OffchainLabs/arbitrum-docs: arbitrum-docs/for-devs/third-party-docs/Tenderly/01-tenderly.mdx

---
title: 'Tenderly Overview'
description: "Tenderly provides full-stack Web3 infrastructure with high-performance Node RPC, mainnet-like development environments, intuitive development and simulation tools, and real-time monitoring for building and scaling secure blockchain applications"
author: 'nvitorovic'
third_party_content_owner: 'nvitorovic'
sidebar_label: 'Overview'
---

# Tenderly: Full Stack Web3 Infrastructure

> [Tenderly](https://tenderly.co/?mtm_campaign=ext-docs&mtm_kwd=arbitrum) enhances development experience on Arbitrum by providing seamless integration with Arbitrum ecosystem. It offers a high-performance Node RPC with integrated mainnet-like development environments, intuitive developer tools, and real-time monitoring.

Reviewer: ```suggestion
[Tenderly](https://tenderly.co) enhances the development experience on Arbitrum by providing seamless integration with the Arbitrum ecosystem. It offers a high-performance Node RPC with integrated mainnet-like development environments, intuitive developer tools, and real-time monitoring.
```

By integrating Tenderly’s full-stack infrastructure, you can **optimize your development workflows** and **scale your dapp** on a high-performance infrastructure.

Explore:
- How to [get started](#get-started)
- Understand Tenderly's key features:
  - [Development & Testing](./02-development-testing.mdx) tools and infrastructure
  - [Node RPC](./04-node-rpc.mdx)
  - [Monitoring & Security](./03-monitoring-security.mdx)
  - [Integration with your workflow](./05-integrating.mdx)
- How you can [integrate in your workflow](./05-integrating.mdx)
- Several [real world examples](#real-world-examples)

## Get Started

> 1. [Create an account](https://dashboard.tenderly.co/register?mtm_campaign=ext-docs&mtm_kwd=arbitrum)

Reviewer: ```suggestion
1. [Create an account](https://dashboard.tenderly.co/register)
```

> 1. [Create an account](https://dashboard.tenderly.co/register?mtm_campaign=ext-docs&mtm_kwd=arbitrum)

Reviewer: ```suggestion
1. [Create an account](https://dashboard.tenderly.co/register)
```

> 2. [Create a virtual TestNet on Arbitrum](https://docs.tenderly.co/virtual-testnets/quickstart?mtm_campaign=ext-docs&mtm_kwd=arbitrum)
> 3. [Use unlimited faucet](https://docs.tenderly.co/virtual-testnets/unlimited-faucet?mtm_campaign=ext-docs&mtm_kwd=arbitrum)
> 4. Learn more on [**official docs**](https://docs.tenderly.co?mtm_campaign=ext-docs&mtm_kwd=arbitrum)

Reviewer: ```suggestion
2. [Create a virtual TestNet on Arbitrum](https://docs.tenderly.co/virtual-testnets/quickstart)
3. [Use unlimited faucet](https://docs.tenderly.co/virtual-testnets/unlimited-faucet)
4. Learn more on [**official docs**](https://docs.tenderly.co)
```

## How to use Tenderly

Web3 teams can use Tenderly’s infrastructure and tools to build and scale secure and high-quality Web3 software faster and more efficiently.

Here are some of the ways you can use Tenderly

> - **Test and stage contracts and dapps**: Use [Virtual TestNets](https://docs.tenderly.co/virtual-testnets/?mtm_campaign=ext-docs&mtm_kwd=arbitrum) in CI/CD pipelines to deploy, test, and stage smart contracts in mainnet-like development environments, eliminating the need for local setups.
> - **Optimize smart contracts**: Debug and optimize transaction inputs using [Simulator](https://docs.tenderly.co/simulator-ui?mtm_campaign=ext-docs&mtm_kwd=arbitrum) and [Debugger](https://docs.tenderly.co/debugger?mtm_campaign=ext-docs&mtm_kwd=arbitrum), reducing debugging time from hours to minutes with visual transaction trace inspection.
> - **Predict transaction outcomes**: Integrate [Simulation RPC](https://docs.tenderly.co/simulations/single-simulations#simulate-via-rpc?mtm_campaign=ext-docs&mtm_kwd=arbitrum) to simulate transaction execution and gas costs, preventing errors and optimizing token transfers before submitting transactions on-chain.
> - **Automate security responses**: Implement automated responses to on-chain events and protocol issues using real-time [Alerts](https://docs.tenderly.co/alerts/intro-to-alerts?mtm_campaign=ext-docs&mtm_kwd=arbitrum) and [Web3 Actions](https://docs.tenderly.co/web3-actions/intro-to-web3-actions?mtm_campaign=ext-docs&mtm_kwd=arbitrum) to safeguard your dapps and protocols.
> - **Ensure high-performance dapp connectivity**: Use [Node RPC](https://docs.tenderly.co/node?mtm_campaign=ext-docs&mtm_kwd=arbitrum) to guarantee low-latency, high-reliability, and multi-region access for your dapp, ensuring smooth operation even during high traffic volumes and spikes.

Reviewer: ```suggestion
- **Test and stage contracts and dapps**: Use [Virtual TestNets](https://docs.tenderly.co/virtual-testnets/) in CI/CD pipelines to deploy, test, and stage smart contracts in mainnet-like development environments, eliminating the need for local setups.
- **Optimize smart contracts**: Debug and optimize transaction inputs using [Simulator](https://docs.tenderly.co/simulator-ui) and [Debugger](https://docs.tenderly.co/debugger), reducing debugging time from hours to minutes with visual transaction trace inspection.
- **Predict transaction outcomes**: Integrate [Simulation RPC](https://docs.tenderly.co/simulations/single-simulations#simulate-via-rpc) to simulate transaction execution and gas costs, preventing errors and optimizing token transfers before submitting transactions on-chain.
- **Automate security responses**: Implement automated responses to on-chain events and protocol issues using real-time [Alerts](https://docs.tenderly.co/alerts/intro-to-alerts) and [Web3 Actions](https://docs.tenderly.co/web3-actions/intro-to-web3-actions) to safeguard your dapps and protocols.
- **Ensure high-performance dapp connectivity**: Use [Node RPC](https://docs.tenderly.co/node) to guarantee low-latency, high-reliability, and multi-region access for your dapp, ensuring smooth operation even during high traffic volumes and spikes.
```
